src/navigator_data_ingest/main.py

```diff
@@ -134,23 +127,13 @@ def main(
     param indexer_input_prefix: S3 prefix containing the indexer input files.
     param archive_prefix: S3 prefix to which to archive documents.
     param worker_count: Number of workers downloading/uploading cached documents.
-    param execution_id: Unique identifier for the execution.
-    param execution_data_prefix: Prefix to apply to output files that contains the
-    execution data files.
+    param db_state_file_key: The s3 path for the file containing the db state
     """
 
-    # Read the execution data file to get the unique execution timestamp s3 path
-    # (input/${timestamp}/)
-    execution_data_path = (
-        S3Path(os.path.join("s3://", pipeline_bucket, execution_data_prefix))
-        / f"{execution_id}.json"
-    )
-
-    input_dir_path = S3Path(
-        ExecutionData.model_validate_json(
-            execution_data_path.read_text()
-        ).input_dir_path
-    )
+    # Get the key of folder containing the db state file
+    input_dir_path = (
+        S3Path(os.path.join("s3://", pipeline_bucket, db_state_file_key))
+    ).parent
```

Comment:
**question:** Why is it the parent, and not the full key?

Reply:
The path is then used to get the updates file. It's a little janky still but I'd see it as an improvement. I did look at passing in the updates file directly instead, but found that it would be tricky from the pipeline repo so went this way instead